app/api/views.py

```diff
@@ -90,9 +90,14 @@ def label_per_data(self, project):
         annotation_class = project.get_annotation_class()
         docs = project.documents.all()
         annotations = annotation_class.objects.filter(document_id__in=docs.all())
-        for d in annotations.values('label__text', 'user__username').annotate(Count('label'), Count('user')):
-            label_count[d['label__text']] += d['label__count']
-            user_count[d['user__username']] += d['user__count']
+        if annotation_class == Seq2seqAnnotation:
```

Comment:
Type-checking in a conditional often is an indicator for a potential refactor using polymorphism. Wondering if this would be clearer as a method on the base Annotation class that gets overriden in the Seq2SeqAnnotation class.

```py
class Annotation(...):
  # ...
  def some_clever_name(self, docs):
    label_count = Counter()
    user_count = Counter()
    annotations = annotation_class.objects.filter(document_id__in=docs.all())
    # ... else block logic here ...

class Seq2SeqAnnotation(Annotation):
  def some_clever_name(self, docs):
    # ... logic for if block here ...
```

[Further reading.](http://wiki.c2.com/?ReplaceConditionalWithPolymorphism)

Reply:
Changed this to use an `AnnotationManager`.